### ISSUE TITLE
stop using web3-provider-engine

### ISSUE THREAD

**Issue by g00dnatur3** [trufflesuite/truffle#4468]
i must see this error 100 times a day, literally!

```
/Users/user/Desktop/_yoer/node_modules/@trufflesuite/web3-provider-engine/index.js:163
        cb(null, null)
        ^
Error: Could not find block
    at /Users/user/Desktop/_yoer/node_modules/@trufflesuite/web3-provider-engine/index.js:62:28
    at afterRequest (/Users/user/Desktop/_yoer/node_modules/@trufflesuite/web3-provider-engine/index.js:163:9)
    at /Users/user/Desktop/_yoer/node_modules/@trufflesuite/web3-provider-engine/index.js:177:12
```

these guys fixed it by stop using web3-provider-engine:

https://github.com/NoahZinsmeister/web3-react/issues/63

mmmmm.... yeah...mmmmmm
if you can fix this

that would be great

**Comment by gnidan** [comment]
We're not altogether too happy about the situation either, I promise ;)

See [this PR](https://github.com/trufflesuite/truffle/pull/4416) to follow along with our efforts to upgrade that package to the upstream latest.

Sorry about the source of frustration here!

**Comment by lebed2045** [comment]
+1 of how annoying is it

**Comment by haltman-at** [comment]
Since #4416 is now merged, could you try updating to latest Truffle and see if the issue still exists?  Thanks!

**Comment by eggplantzzz** [comment]
Can anyone confirm/deny that this is a issue on the latest Truffle (currently 5.4.29)?

**Comment by MicaiahReid** [comment]
Closing for issue maintenance. Please reopen or make another issue if this is still happening after version 5.4.29.

**Comment by janezkranjc** [comment]
This is still an issue with latest truffle.

**Comment by cds-amal** [comment]
@janezkranjc can you confirm the truffle version? `truffle version` ?

**Comment by janezkranjc** [comment]
```
➜  blockchain git:(master) truffle version
Truffle v5.5.0 (core: 5.5.0)
Ganache v7.0.1
Solidity v0.5.16 (solc-js)
Node v17.5.0
Web3.js v1.5.3
```

**Comment by cds-amal** [comment]
@vipulw2011 can you specify your truffle, node and npm versions? (`truffle version` and `npm --version`)? If you're at v5.5.0, could you please open a new issue, so we can track and get resources on it?

> I'm also facing the same issue and we are suppose to go live today. Attaching the error screenshot for any solutions.

Also, if by live you mean you need to deploy your contracts to main net, might I suggest `Truffle dashboard` released with Truffle v5.5.0, which allows you to publish your contracts without sharing your mnemonic with truffle, and instead, rely on MetaMask to manage signing. See the [release notes for more information,](https://github.com/trufflesuite/truffle/releases/tag/v5.5.0) or check out [this video demonstration.](https://www.youtube.com/watch?v=AngblG9TY18)

**Comment by kimyd** [comment]
My truffle verison is 

Truffle v5.5.1 (core: 5.5.1)
Ganache v7.0.1
Solidity - 0.8.0 (solc-js)
Node v16.13.1
Web3.js v1.5.3

But still I got the error when I start up run script to Mainnet.

**Comment by jsgphd** [comment]
I am having this issue as well, I deploy to polygon_infura_testnet fine, but on polygon_infura_mainnet, even with gasLimit: 3000000000

My truffle version is:
Truffle v5.5.1 (core: 5.5.1)
Ganache v7.0.1
Solidity - 0.8.12 (solc-js)
Node v16.8.0
Web3.js v1.5.3

PLEASE HELP.  I am totaly blocked on this.

**Comment by wpMcDonnell** [comment]
Same I have been dealing with this for a week now. I can console in to any network, I can deploy on all other tests nets, just not Matic Polygon... i had some luck retrying over and over a week ago, but essentially ruined my deployments where it errored out last second before saving my migrated contracts

Truffle v5.3.0 (core: 5.3.0)
Solidity v0.5.16 (solc-js) but really v0.8.11
Node v16.14.0
Web3.js v1.2.9

**Comment by jsgphd** [comment]
Can you guys re-open this issue.  It's obviouly a huge problem for anyone using polygon mainnet via truffle and appears to have no work arounds.

**Comment by kimyd** [comment]
I think this issue is related to Infura RPC endpoint. After I changed to Alchemy RPC endpoint, it works fine. Nothing wrong with truffle.

**Comment by wpMcDonnell** [comment]
Can confirm @kimyd , alchemy RPC endpoint fixed it. Deployed fast, also their endpoint manager and site is legit. Thanks!!!!
